Source/CBL_Attachment.m

```diff
@@ -101,8 +101,13 @@ - (instancetype) initWithName: (NSString*)name
             // Get the revpos:
             id revPosObj = attachInfo[@"revpos"];
             if (revPosObj) {
-                int revPos = [$castIf(NSNumber, revPosObj) intValue];
-                if (revPos <= 0) {
+                if (![revPosObj isKindOfClass: [NSNumber class]]) {
+                    *outStatus = kCBLStatusBadAttachment;
+                    return nil;
+                }
+                int revPos = [revPosObj intValue];
+                // PouchDB has a bug that generates "revpos":0; allow this (#1200)
+                if (revPos < 0) {
```

Comment:
It's already possible for an incoming doc attachment to end up with a revpos of 0, if its attachment dict doesn't have a `revpos` property at all. The error case I'm fixing is one where there _is_ a `revpos` but its value is 0. So I don't think this patch allows any new conditions that we didn't get before.